### PR TITLE
add resume for dflash training script

### PR DESCRIPTION
**Pull request by tianhaoz95**
## Motivation

The current dflash saves checkpoint but does not have logic to resume training, dflash tend to have a higher epoch count compared with eagle3, so this will be helpful.

## Modifications

* change the save checkpoint to be on the epoch resolution since we don't have logic to restore optimizer state
* when loading drafter model, use checkpoint if requested with `--resume` and available

## Related Issues

NA

## Accuracy Test

The loading and training proceeds:

```bash
INFO:specforge.utils:/tmp/tianhaoz/longcat_dflash_training/outputs
INFO:specforge.utils:Last checkpoint detected: /tmp/tianhaoz/longcat_dflash_training/outputs/epoch_4
INFO:specforge.utils:Loaded draft config from checkpoint: /tmp/tianhaoz/longcat_dflash_training/outputs/epoch_4
```

## Benchmark & Profiling

This change is unrelated to performance.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

File: scripts/train_dflash.py [hunk 1]
```diff
@@ -96,7 +96,6 @@ def parse_args():
     output_group.add_argument("--cache-dir", type=str, default="./cache")
     output_group.add_argument("--log-interval", type=int, default=50)
     output_group.add_argument("--eval-interval", type=int, default=1000)
-    output_group.add_argument("--save-interval", type=int, default=1000)
```
Comment (FrankLeeeee) [inline scripts/train_dflash.py:123]: I think we can keep this argument for consistency.

Comment (tianhaoz95) [reply scripts/train_dflash.py:123]: sounds good. added it back
